File: NodeJS/GetTranslationsArray.js
'use strict';

let https = require ('https');

// **********************************************
// *** Update or verify the following values. ***
// **********************************************

// Replace the subscriptionKey string value with your valid subscription key.
let subscriptionKey = 'ENTER KEY HERE';

let host = 'api.microsofttranslator.com';
let path = '/V2/Http.svc/GetTranslationsArray';

let params = '';

let from = "en-us";
let to = "fr-fr";

let ns = "http://schemas.microsoft.com/2003/10/Serialization/Arrays";
let content =
	"<GetTranslationsArrayRequest>" +
	"  <AppId />" +
	"  <From>" + from + "</From>" +
	"  <Texts>" +
	"    <string xmlns=\"" + ns + "\">Hello</string>" +
	"    <string xmlns=\"" + ns + "\">Goodbye</string>" +
	"  </Texts>" +
	"  <To>" + to + "</To>" +
	"  <MaxTranslations>10</MaxTranslations>" +
	"</GetTranslationsArrayRequest>";

let response_handler = function (response) {
    let body = '';
    response.on ('data', function (d) {
        body += d;
    });
    response.on ('end', function () {
		console.log (body);
    });
    response.on ('error', function (e) {
        console.log ('Error: ' + e.message);
    });
};

let GetTranslationsArray = function () {
	let request_params = {
		method : 'POST',
		hostname : host,
		path : path + params,
		headers : {
			'Content-Type' : 'text/xml',
			'Ocp-Apim-Subscription-Key' : subscriptionKey,
		}
	};

	let req = https.request (request_params, response_handler);
	req.write (content);
	req.end ();
}

GetTranslationsArray ();

File: Ruby/GetTranslationsArray.rb
require 'net/https'
require 'uri'
require 'cgi'

# **********************************************
# *** Update or verify the following values. ***
# **********************************************

# Replace the key string value with your valid subscription key.
key = 'ENTER KEY HERE'

host = 'https://api.microsofttranslator.com'
path = '/V2/Http.svc/GetTranslationsArray'

params = ''

from = "en-us";
to = "fr-fr";

ns = "http://schemas.microsoft.com/2003/10/Serialization/Arrays";
body =
	"<GetTranslationsArrayRequest>" +
	"  <AppId />" +
	"  <From>" + from + "</From>" +
	"  <Texts>" +
	"    <string xmlns=\"" + ns + "\">Hello</string>" +
	"    <string xmlns=\"" + ns + "\">Goodbye</string>" +
	"  </Texts>" +
	"  <To>" + to + "</To>" +
	"  <MaxTranslations>10</MaxTranslations>" +
	"</GetTranslationsArrayRequest>";

uri = URI (host + path + params)

request = Net::HTTP::Post.new(uri)
request['Content-type'] = 'text/xml'
request['Ocp-Apim-Subscription-Key'] = key
request.body = body

response = Net::HTTP.start(uri.host, uri.port, :use_ssl => uri.scheme == 'https') do |http|
    http.request (request)
end

puts response.body
